src/Serializer/CustomContextBuilder.php

```diff
@@ -58,10 +61,47 @@ private function extractPathParameters(Request $request, array &$context): void
             ARRAY_FILTER_USE_KEY
         );
 
-        $context['path_parameters'] = array_map(
-            static fn (string $pathParameter) => StringUtil::castFromString($pathParameter),
-            $pathParameters
-        );
+        /**
+         * @var string $pathParameterName
+         * @var string $pathParameterValue
+         */
+        foreach ($pathParameters as $pathParameterName => $pathParameterValue) {
+            /** @var class-string $resourceClass */
+            $resourceClass = $context['resource_class'];
+
+            $propertyType = $this->extactPropertyTypeFromResourceClass($resourceClass, $pathParameterName);
+
+            if ($propertyType !== null) {
+                settype($pathParameterValue, $propertyType);
+            } else {
+                $pathParameterValue = StringUtil::castFromString($pathParameterValue);
+            }
+
+            $pathParameters[$pathParameterName] = $pathParameterValue;
+        }
+
+        $context['path_parameters'] = $pathParameters;
+    }
+
+    /**
+     * @param class-string $resourceClass
+     */
+    private function extactPropertyTypeFromResourceClass(string $resourceClass, string $propertyName): ?string
+    {
+        $resourceClassInstance = (new ReflectionClass($resourceClass))->newInstanceWithoutConstructor();
+        if (! $resourceClassInstance instanceof EventEngineType) {
+            return null;
+        }
+
+        $properties = $resourceClassInstance
```

Comment:
We can already get this properties outside the foreach statement in the previous method. So we don't need to fetch the resourc calss, create an instance of it and fetch the schema for every parameter.

Reply:
If we keep that in mind we don't need this extactPropertyTypeFromResourceClass function: `$properties[StringUtil::camelize($propertyName)]['type'] ?? null` will also do the trick

Reply:
I moved `newInstanceWithoutConstructor` outside the `foreach`, but I prefer to keep the function extactPropertyTypeFromResourceClass as it's easier to use it if resourceClass is not an instance of EventEngineType